core/src/main/kotlin/net/corda/core/serialization/amqp/custom/ThrowableSerializer.kt

```
@@ -0,0 +1,78 @@
+        val message = if (obj is CordaThrowable) {
+            // Try and find a constructor
+            try {
+                val `constructor` = constructorForDeserialization(obj.javaClass)
+                val props = propertiesForSerialization(`constructor`, obj.javaClass, factory)
+                for (prop in props) {
+                    extraProperties[prop.name] = prop.readMethod.invoke(obj)
+                }
+            } catch(e: NotSerializableException) {
+            }
+            obj.originalMessage
+        } else {
+            obj.message
+        }
+        return ThrowableProxy(obj.javaClass.name, message, obj.stackTrace, obj.cause, obj.suppressed, extraProperties)
+    }
+
+    override fun fromProxy(proxy: ThrowableProxy): Throwable {
+        try {
+            // TODO: This will need reworking when we have multiple class loaders
+            var clazz = Class.forName(proxy.exceptionClass, false, this.javaClass.classLoader)
+            // If it is CordaException or CordaRuntimeException, we can seek any constructor and then set the properties
+            // Otherwise we just make a CordaRuntimeException
+            if (CordaThrowable::class.java.isAssignableFrom(clazz) && Throwable::class.java.isAssignableFrom(clazz)) {
+                val `constructor` = constructorForDeserialization(clazz)!!
+                return ((`constructor`.callBy(`constructor`.parameters.map { it to proxy.additionalProperties[it.name] }.toMap()) as CordaThrowable).apply {
+                    if (this.javaClass.name != proxy.exceptionClass) this.originalExceptionClassName = proxy.exceptionClass
+                    this.setMessage(proxy.message)
+                    this.setCause(proxy.cause)
+                    this.addSuppressed(proxy.suppressed)
+                } as Throwable).apply {
+                    this.stackTrace = proxy.stackTrace
+                }
+            }
+        } catch (e: Exception) {
```

At least a comment as why the exception is ignored